### DISCUSSION TITLE
Are unencrypted files ever written to disk?

### DISCUSSION THREAD

**Discussion by Kreonid** [pyinstaller/pyinstaller#7292] [Help]
I want to know if the files protected by the PyInstaller's encryption (`--key` flag) are ever written to disk unencrypted? I tested this with my app and it seems only pyd and dll files are written to the _MEIXXX directory, and not the py (pyc) files of my script. Could you confirm?

**Comment by bwoodsend** [comment]
No but `--key` in not secure so don't use it. See #6999.

**Reply by Kreonid** [reply]
Thanks. I also have another question about encryption. 

I tried to alter the encryption a little for exactly the reasons you describe in that issue. However, I'm getting an error when starting my script.

**The error:**
```
Traceback (most recent call last):
      File "PyInstaller\loader\pyimod03_importers.py", line 298, in get_code
      File "PyInstaller\loader\pyimod02_archive.py", line 298, in extract
    zlib.error: Error -3 while decompressing data: incorrect header check

    The above exception was the direct cause of the following exception:

    Traceback (most recent call last):
      File "PyInstaller\loader\pyiboot01_bootstrap.py", line 25, in <module>
      File "PyInstaller\loader\pyimod03_importers.py", line 478, in exec_module
      File "PyInstaller\loader\pyimod03_importers.py", line 300, in get_code
    ImportError: Loader FrozenImporter cannot handle module os
    [8172] Failed to execute script 'pyiboot01_bootstrap' due to unhandled exception!
```

**What I changed:**
I added a line `key = "123" + key` after [this line](https://github.com/pyinstaller/pyinstaller/blob/15f23a8a89be5453b3520df8fc3e667346e103a6/PyInstaller/archive/pyz_crypto.py#L22) and after [this line](https://github.com/pyinstaller/pyinstaller/blob/7875d7684c6ddad90e91af9141229083ebb1297f/PyInstaller/loader/pyimod01_archive.py#L214).

**My understanding**
These are two classes that are used for encryption and decryption of the python code. The crypto key written to the PyInstaller archive will not have a "123" prefix, but the prefix will be added at the runtime. Could you suggest why it doesn't work?

**Reply by rokm** [reply]
> Could you suggest why it doesn't work?

Try adding a `print(key)` before you modify it [in the second file](https://github.com/pyinstaller/pyinstaller/blob/7875d7684c6ddad90e91af9141229083ebb1297f/PyInstaller/loader/pyimod01_archive.py#L214), and you'll see why this doesn't work...

Basically, your first change is practically the same as changing the initial key on the command-line or in the spec file (because the change is propagated into generated `build/<name>/pyimod00_crypto_key.py`), while the second change is therefore applied to already-modified (and zero-padded) key.

**Reply by Kreonid** [reply]
Thank you! I've solved my issue.
